### PR TITLE
Quickfix the position to get godot editor top buttons

### PR DESCRIPTION
**Pull request by Meta-Ben**
### Description

This pull request addresses an issue on macOS where the window "open/resize/close" buttons are positioned on the left side, causing a modification in the order of the Godot top view buttons (like the "2D/3D/Etc" buttons). The fix ensures that the position of these top buttons is adapted specifically for macOS.

### Changes

Added a function to get the position of the top buttons that takes into account the macOS operating system.
Integrated this function into the `get_top_buttons()` function to adjust the button positions appropriately.

### How to Test

- Open the Godot editor on a macOS system.
- Verify that there are no errors such as:
`res://addons/fabimakesgames.embed_game/embed_game.gd:122 - Invalid assignment of property or key 'shortcut' with value of type 'Shortcut' on a base object of type 'Nil'.` or 
`res://addons/fabimakesgames.embed_game/embed_game.gd:140 - Invalid access to property or key 'button_pressed' on a base object of type 'Nil'.`
- Check if the plugin works correctly when toggling the Embed top button and running your game.
- Test the plugin on other operating systems (e.g., Windows) to ensure the changes do not affect the way the positions of the top buttons are determined.

### Additional Notes

This fix specifically targets macOS and is fully written in GDScript.

### CODE REVIEW

File: addons/fabimakesgames.embed_game/embed_game.gd
Comment (cyanglaz) [inline addons/fabimakesgames.embed_game/embed_game.gd:254]: I wonder if there is a more robust way to do this, for example if mac or any OS changes their button layout this will fail.
Is there a way to identify the container? If there is, we can do a loop and search for that specific container.

Comment (Meta-Ben) [reply addons/fabimakesgames.embed_game/embed_game.gd:254]: Totally agree ( same for linux user that can have the mac buttons or the windows buttons even in the middle for certain distrib ) 

I made this as a quickfix to allow macOs contributors/users to quickly come into the plugin, in fact we can identify the tops buttons by the name of the HBox used, but same, if godot change the name of this hbox it will also fail, both have potential fail situation ( but I think that mac OS test was a "longer" quick fix than Godot naming, since Godot make lot of work in depth for such thing than Apple Developer on macOs )

maybe there is another solution but I didn't explore more for now

Comment (nan0m) [reply addons/fabimakesgames.embed_game/embed_game.gd:254]: Makes me wonder if ultimately a different way  of adding the button should be implemented. As it is kinda hacky to just add HBOXcontainer + button to the 2D/3D/Script/Asset Store buttons.
But I thought it was quite appealing UX to have that button there

Comment (nan0m) [reply addons/fabimakesgames.embed_game/embed_game.gd:254]: > I wonder if there is a more robust way to do this, for example if mac or any OS changes their button layout this will fail.
> Is there a way to identify the container? If there is, we can do a loop and search for that specific container.

Potentially we could look for one of the other buttons. 3D,2D, Assets, or Script and then get their parent. The Buttons should have their respective names in their text property. So that's what we could check against

Comment (Meta-Ben) [reply addons/fabimakesgames.embed_game/embed_game.gd:254]: Maybe but the top bar seem a nice place !

Godot contributors actually work on the same feature for the engine : https://github.com/godotengine/godot/pull/97257

And they put it on the top bar too, so If we want a more robust way to put the button on top maybe you can check the code they made to add the button on the top bar ( I think it's directly in C++ and this PR target the gdscript only branch )

I wonder if it can be a good idea to have the button directly next to play ( to let the user choose if he want to play in editor or if he want to use the regular play button that open a window )

Comment (cyanglaz) [reply addons/fabimakesgames.embed_game/embed_game.gd:254]: I wouldn't block on this PR tho. I think it would be great if we can land this to unblock mac users. We can always refine later.

Comment (Meta-Ben) [reply addons/fabimakesgames.embed_game/embed_game.gd:254]: Thanks I agree with you ! @nan0m I let you merge the PR if you are ok with that too :)

Comment (nan0m) [reply addons/fabimakesgames.embed_game/embed_game.gd:254]: @Meta-Ben I agree. I will merge for now, thank you!

**Review by nan0m — APPROVED** [review]

**Pull request merged**
